src/validation/schema.js

```diff
@@ -322,13 +378,200 @@ const validateEntityFields = (defs, def) =>
+      }),
+    )
+  }
+
+  if (argument.value.fields.length != 1) {
+    return List().push(
+      immutable.fromJS({
+        loc: directive.name.loc,
+        entity: def.name.value,
+        message: `@imports directive argument: 'from' must have one field of id or name`,
+      }),
+    )
+  }
+
+  return argument.value.fields.reduce((errors, field) => {
+    if (!['name', 'id'].includes(field.name.value)) {
+      return errors.push(
+        immutable.fromJS({
+          loc: field.name.loc,
+          entity: def.name.value,
+          message: `@imports directive from: only fields 'id' or 'name' allowed`,
+        }),
+      )
+    }
+    if (field.value.kind != 'StringValue') {
+      return errors.push(
+        immutable.fromJS({
+          loc: field.name.loc,
+          entity: def.name.value,
+          message: '@imports directive from: fields must have string values',
+        }),
+      )
+    }
+    return errors
+  }, List())
+}
+
+const validateImportDirectiveTypes = (def, directive) => {
+  let types = directive.arguments.find(argument => argument.name.value == 'types')
+  return types
+    ? validateImportDirectiveArgumentTypesIsValid(def, directive, types)
+    : List([
+        immutable.fromJS({
+          loc: directive.name.loc,
+          entity: def.name.value,
+          message: `@import argument 'types' must be specified`,
+        }),
+      ])
+}
+
+const validateImportDirectiveFrom = (def, directive) => {
+  let from = directive.arguments.find(argument => argument.name.value == 'from')
+  return from
+    ? validateImportDirectiveArgumentFromIsValid(def, directive, from)
+    : List([
+        immutable.fromJS({
+          loc: directive.name.loc,
+          entity: def.name.value,
+          message: `@import argument 'from' required: specify the subgraph to import types from`,
+        }),
+      ])
+}
+
+const validateImportDirectiveName = directive =>
+  directive.name.value != 'imports'
+    ? List([
+        immutable.fromJS({
+          loc: directive.name.loc,
+          entity: def.name.value,
+          message: `${SUBGRAPH_SCHEMA} directives: only @import directives allowed`,
+        }),
+      ])
+    : List()
+
+const validateImportDirective = (def, directive) =>
+  List.of(
+    ...validateImportDirectiveName(directive),
+    ...validateImportDirectiveTypes(def, directive),
+    ...validateImportDirectiveFrom(def, directive),
+  )
+
+const validateSubgraphSchemaDirectives = def =>
+  def.directives.reduce(
+    (errors, directive) => errors.concat(validateImportDirective(def, directive)),
+    List(),
+  )
+
+const validateTypeHasNoFields = def =>
+  def.fields.length
+    ? List([
+        immutable.fromJS({
+          loc: def.name.loc,
+          entity: def.name.value,
+          message: `${def.name.value} type is not allowed any fields by convention`,
+        }),
+      ])
+    : List()
+
+const validateAtLeastOneExtensionField = def => List()
```

Comment:
I assume this is a placeholder for the `extend type` phase of this feature implementation?

Reply:
Indeed